### DISCUSSION TITLE
Occured error when follow your instructions.

### DISCUSSION THREAD

**Discussion by ityspace** [ryan4yin/nixos-and-flakes-book#27] [Q&A] [answered]
```
let
  a = 1;
in
"the value of a is ${a}" 
```

THEN output that :
error: cannot coerce an integer to a string

       at «string»:4:2:

            3| in
            4| "the value of a is ${a}"
             |                    ^
            5|

**Comment by ityspace** [comment]
So the interpolation cannot be an integer？

**Comment by ryan4yin** [comment, marked as answer]
Yep, I made a mistake here, I found an issue related to it: https://github.com/NixOS/nix/issues/7327

**Comment by ryan4yin** [comment]
Fixed, I changed it to `a = "1";`
